### PR TITLE
[improve][pip] PIP-380: Support setting up specific namespaces to skipping the load-shedding

### PR DESCRIPTION
**Pull request by Demogorgon314**

### CODE REVIEW

**Comment by lhotari** [comment]
It seems fine. Just thinking that there could be similar requirements for user defined namespaces and therefore having a property in the namespace policy to control this would be useful. Instead of a single binary value, perhaps it could be some type of value that would be used for ordering the namespace bundles that are picked as candidates for load shedding. 
A system namespace could use a value that would result in all other namespace bundles being picked before it becomes a candidate for load shedding.

**Comment by Demogorgon314** [comment]
@lhotari I'm a little worried that adding a new namespace policy may make the concept too complex, and if we truly want a namespace policy we can make another PIP for that future, right?

**Comment by heesung-sn** [comment]
I wonder if we can limit this blacklisting to the TransferShedder under `loadBalanceTransferShedderSheddingExcludedNamespaces` config .

**Comment by Demogorgon314** [comment]
@heesung-sn I would like to use the `loadBalancerSheddingExcludedNamespaces` config to support `ExtensibleLoadManager` and `ModularLoadManager`

**Comment by heesung-sn** [comment]
I think we probably want to explore namespace policy like this work.

https://github.com/apache/pulsar/pull/23120

**Comment by Demogorgon314** [comment]
Adding a namespace policy is a good idea, let me update the PIP.

**Comment by Demogorgon314** [comment]
@lhotari Are you suggesting adding a `loadSheddingPriority` to the namespace police? I'm wondering which case will use the priority for load shedding. And when using priority for each namespace will make the load-shedding strategy let bit complex, since when we select the hot bundle to unload we have to consider the namespace priority. I prefer to use  `loadSheddingEnabled`, then for the protocol's system topic and some user's benchmark topic we can make the ownership stable.

**Comment by Demogorgon314** [comment]
ping @lhotari

File: pip/pip_379.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
+# PIP-379: Support setting up specific namespaces to skipping the load-shedding
```
Comment (lhotari) [inline pip/pip_379.md:1]:
```suggestion
# PIP-380: Support setting up specific namespaces to skipping the load-shedding
```

**Comment by lhotari** [comment]
PIP-379 is already taken, I increased the number to PIP-380. The way to find the next available number is to check the dev mailing list and open PRs. The first one to open a discussion for a specific PIP number takes it. This is the PIP-379 discussion: https://lists.apache.org/thread/l5zjq0fb2dscys3rsn6kfl7505tbndlx

**Review by lhotari — APPROVED** [review]
LGTM

**Comment by lhotari** [comment]
> @lhotari Are you suggesting adding a `loadSheddingPriority` to the namespace police? I'm wondering which case will use the priority for load shedding. And when using priority for each namespace will make the load-shedding strategy let bit complex, since when we select the hot bundle to unload we have to consider the namespace priority. I prefer to use `loadSheddingEnabled`, then for the protocol's system topic and some user's benchmark topic we can make the ownership stable.

@Demogorgon314 It was something to consider for the future.

**Comment by lhotari** [comment]
@Demogorgon314  Please update the links to the PIP discussion and voting threads.

**Review by RobertIndie — APPROVED** [review]

**Review by heesung-sn — APPROVED** [review]

**Pull request merged**